tests/core_test.py

```diff
@@ -431,17 +431,17 @@ def new_jaxpr():
     jaxpr.eqns[0].outvars[0].aval = make_shaped_array(jnp.int32(2))
     self.assertRaisesRegex(
         core.JaxprTypeError,
-        r"Variable '.' inconsistently typed as ShapedArray(.*), "
-        r"bound as ShapedArray(.*)\n\nin equation:\n\n.:i32\[\] = sin .",
+        r"Variable 'b' inconsistently typed as f32\[\], "
+        r"bound as i32\[\]\n\nin equation:\n\nb:i32\[\] = sin a",
```

Comment:
I added tests! Well actually I just added variable names to the existing test. This fails on the main branch because of exactly the issue that this PR fixes.